### ISSUE TITLE
Build: Version comes from git tag, ChangeLog, Jamulus.pro

### ISSUE THREAD

**Issue by hoffie** [jamulussoftware/jamulus#2662]
> My concern is `get_build_version()` uses a different value.  The tag, `Jamulus.pro` and `ChangeLog` could all have different versions.

> Given we have three sources for the version, we really need automation to make sure they're aligned, at least: i.e. error out if things look wrong.  But I agree, that's not in scope here.  For that, it sounds like (and I don't fully understand how this works):
> * If the tag is set, we should use that to determine build type and version and that source should overrule others.
> * ChangeLog and Jamulus.pro "target" versions should match - and, if set, match the tag build type and version.
> * If the tag wasn't set, build type and version can then come from ChangeLog and Jamulus.pro, as needed.

_Originally posted by @pljones in https://github.com/jamulussoftware/jamulus/pull/2656#discussion_r901061485_

**Comment by ann0see** [comment]
Just a note, we shouldn't extract the version from ChangeLog.
We could automatically extract the version from the tag and put it into Jamulus.pro

**Comment by pljones** [comment]
> Just a note, we shouldn't extract the version from ChangeLog. We could automatically extract the version from the tag and put it into Jamulus.pro

I think the problem is that there might not be a useful tag when someone triggers the autobuild -- and it still has to work.  So it has to get a version from somewhere.  To me, `Jamulus.pro` should actually be the overruling version (regardless of tag).  That _must_ exist for a run to work.  ChangeLog doesn't have to exist for `qmake && make` to work but Jamulus.pro must.

(So yes, I said let the tag overrule -- but I'm not fixated on that...)

**Comment by pljones** [comment]
I don't think we know enough about what the solution will look like to target a release.  Dropping the milestone and moving to Triage.

On another note:
> I think we should write a release script doing the tagging,...

That's too late.  The _build_ needs it at compile time to put into the copyright messages.

**Comment by mcfnord** [comment]
MY LLM WROTE:

I checked whether the three version sources can silently diverge. On `main` they agree and are bumped together, so the procedure works — but three things would let a mismatch through:

**1. Pre-release tags skip the check.** `get_build_vars.py` raises on a tag↔`Jamulus.pro` mismatch only when `is_prerelease` is false. Against today's tree, `refs/tags/r3_12_3` → `ValueError`; `refs/tags/r4_0_0beta1` → exit 0.

**2. Two releases already shipped through that gap** — `r3_11_0beta1` (`.pro` = `3.11.0beta`) and `r3_9_0nightly1` (`3.9.0Nightly1`), both pre-releases. No non-pre-release tag has ever mismatched.

**3. The ChangeLog guard matches any section, not the newest.** `extractVersionChangelog.pl ChangeLog 3.11.0` exits 0 and returns the 2024 notes, so a stale `.pro` at release time would publish the wrong notes rather than fail the build. `286191ae` ("Fix wrong version in pro file", one day after 3.12.2 shipped) is exactly that case — caught by eye, not CI.

Happy to put numbers behind whichever direction you prefer, or to leave it if the pre-release exemption is deliberate. Reproduction script available.
